primitives/arithmetic/src/per_things.rs

```diff
@@ -30,29 +30,55 @@ pub trait PerThing: Sized + Saturating + Copy {
 	/// The data type used to build this per-thingy.
 	type Inner: BaseArithmetic + Copy;
 
-	/// accuracy of this type
-	const ACCURACY: Self::Inner;
+	/// A data type larger than `Self::Inner`, used to avoid overflow in some computations.
```

Comment:
```suggestion
	/// A data type larger than `Self::Inner`, used to avoid overflow in some computations.
	/// It must be able to compute `ACCURACY^2`.
```

This is needed in the implementation of `fn square`